### ISSUE TITLE
Implement POLY faucet on Kovan

### ISSUE THREAD

**Issue by PolymathTeam** [PolymathNetwork/polymath-issuer#76]
Work includes:
1) solution design
2) build

**Comment by PolymathTeam** [comment]
➤ Stanislav Timofeev commented:

Design: https://invis.io/VTFW8IJEB5M#/304265092_Token_Symbol_Registration-Poly-Modal

**Comment by PolymathTeam** [comment]
➤ Pablo Ruiz commented:

PolyToken on Kovan (which is a ERC20 token with a getTokens method) address is 0x455Da7D06862Fa7d7639473F287f88bc7b35FF7F

**Comment by PolymathTeam** [comment]
➤ Jose Herrera commented:

Is there a contract already that sends out the 25k POLY? Or is that part of this task as well?

**Comment by PolymathTeam** [comment]
➤ Pablo Ruiz commented:

This task is about allowing the issuer to click on a button that calls getTokens on the token contract asking for 25k poly. 
This should only be available on testnet of course as we can't mint POLY from thin air on mainnet :p

**Comment by PolymathTeam** [comment]
➤ Pablo Ruiz commented:

This is the ABI of the faucet deployed at the address I shared:
```
[
 {
 "constant": true,
 "inputs": [],
 "name": "name",
 "outputs": [
 {
 "name": "",
 "type": "string"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 },
 {
 "constant": true,
 "inputs": [],
 "name": "decimals",
 "outputs": [
 {
 "name": "",
 "type": "uint8"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 },
 {
 "constant": true,
 "inputs": [],
 "name": "symbol",
 "outputs": [
 {
 "name": "",
 "type": "string"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 },
 {
 "anonymous": false,
 "inputs": [
 {
 "indexed": true,
 "name": "_from",
 "type": "address"
 },
 {
 "indexed": true,
 "name": "_to",
 "type": "address"
 },
 {
 "indexed": false,
 "name": "_value",
 "type": "uint256"
 }
 ],
 "name": "Transfer",
 "type": "event"
 },
 {
 "anonymous": false,
 "inputs": [
 {
 "indexed": true,
 "name": "_owner",
 "type": "address"
 },
 {
 "indexed": true,
 "name": "_spender",
 "type": "address"
 },
 {
 "indexed": false,
 "name": "_value",
 "type": "uint256"
 }
 ],
 "name": "Approval",
 "type": "event"
 },
 {
 "constant": false,
 "inputs": [
 {
 "name": "_amount",
 "type": "uint256"
 }
 ],
 "name": "getTokens",
 "outputs": [
 {
 "name": "",
 "type": "bool"
 }
 ],
 "payable": false,
 "stateMutability": "nonpayable",
 "type": "function"
 },
 {
 "constant": false,
 "inputs": [
 {
 "name": "_to",
 "type": "address"
 },
 {
 "name": "_value",
 "type": "uint256"
 }
 ],
 "name": "transfer",
 "outputs": [
 {
 "name": "",
 "type": "bool"
 }
 ],
 "payable": false,
 "stateMutability": "nonpayable",
 "type": "function"
 },
 {
 "constant": false,
 "inputs": [
 {
 "name": "_from",
 "type": "address"
 },
 {
 "name": "_to",
 "type": "address"
 },
 {
 "name": "_value",
 "type": "uint256"
 }
 ],
 "name": "transferFrom",
 "outputs": [
 {
 "name": "",
 "type": "bool"
 }
 ],
 "payable": false,
 "stateMutability": "nonpayable",
 "type": "function"
 },
 {
 "constant": true,
 "inputs": [
 {
 "name": "_owner",
 "type": "address"
 }
 ],
 "name": "balanceOf",
 "outputs": [
 {
 "name": "balance",
 "type": "uint256"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 },
 {
 "constant": false,
 "inputs": [
 {
 "name": "_spender",
 "type": "address"
 },
 {
 "name": "_value",
 "type": "uint256"
 }
 ],
 "name": "approve",
 "outputs": [
 {
 "name": "",
 "type": "bool"
 }
 ],
 "payable": false,
 "stateMutability": "nonpayable",
 "type": "function"
 },
 {
 "constant": true,
 "inputs": [
 {
 "name": "_owner",
 "type": "address"
 },
 {
 "name": "_spender",
 "type": "address"
 }
 ],
 "name": "allowance",
 "outputs": [
 {
 "name": "remaining",
 "type": "uint256"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 },
 {
 "constant": true,
 "inputs": [],
 "name": "totalSupply",
 "outputs": [
 {
 "name": "",
 "type": "uint256"
 }
 ],
 "payable": false,
 "stateMutability": "view",
 "type": "function"
 }
 ]
```

**Comment by PolymathTeam** [comment]
➤ Jose Herrera commented:

I'm currently blocked because there seems to be a smart contract mismatch that's preventing my TX to getTokens() from working.

I noticed that the issuer app is using the same  deployed contract that [Pablo Ruiz](https://app.asana.com/0/511546395238232/list) mentioned, 0x455Da7D06862Fa7d7639473F287f88bc7b35FF7F. However, the contract .sol file seems to be mismatched because the TX always fails (Metamask predicts a TX failure and indeed it does). I even copied the contract ("PolyTokenFaucet.sol" under /polymath-issuer/node_modules/polymath-core/contracts) into remix and it fails as well there.

I know there is a mismatch because I used an older contract (one that I actually can't find anymore... not sure where I got it from) called PolyToken.sol and it was able to give me testnet POLY via remix.

This contract has the signature: 
contract PolyToken is IERC20 {

**Comment by PolymathTeam** [comment]
➤ Jose Herrera commented:

[Pablo Ruiz](https://app.asana.com/0/511546395238232/list) mentioned a way to get around this, will try it out tomorrow.

**Comment by PolymathTeam** [comment]
➤ Boris Shevchenko commented:

Bug fixed in polymath-core@1.2.1, thanks [Jose Herrera](https://app.asana.com/0/724140591411017/list) for research and to [Satyam](https://app.asana.com/0/513026740476812/list) for fix
But [Pablo Ruiz](https://app.asana.com/0/511546395238232/list) still should run some tx on kovan to fix ticker reservation (and probably something else as well), now transactions fails with no error message.
Reassigning issue to me since turned out from the [Jose Herrera](https://app.asana.com/0/724140591411017/list) research that this issue (together with accomplished [UI-38 ⁃ Create ConfirmModal and AlertModal component for Polymath-ui](https://app.asana.com/0/0/719932910699161)) is so simple that I'll accomplish it by testing new polymath.js version on polymath-issuer (since we don't have any unit tests in polymath.js yet)
